### PR TITLE
perf: profile TransactionsManager::poll hash fetching

### PR DESCRIPTION
**Pull request by Elvis339**
Issue: https://github.com/paradigmxyz/reth/issues/12842

This PR attempts to profile TransactionsManager::poll hash fetching.

### CODE REVIEW

**Comment by Elvis339** [comment]
Test case added for debugging purposes, I will delete it later.

File: crates/net/network/src/test_only.rs
Comment (mattsse) [inline crates/net/network/src/test_only.rs:18]: I'd actually like to keep this but ignore before we merge, I have a feeling this will be useful again

File: crates/net/network/src/test_only.rs
Comment (mattsse) [inline crates/net/network/src/test_only.rs:1]: can we move this to the tests/it/ folder or does this require private API access?

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, I like this a lot.

testing this one specific code path is a bit challenging, I'll also try this out.

featurewise, I consider this PR completed once it compiles, because the next part is now just debugging work.

File: crates/net/network/src/test_only.rs
Comment (Elvis339) [reply crates/net/network/src/test_only.rs:1]: Sure! Moved, thank you for the help & review. I'll try to have profiling data asap.

File: crates/net/network/tests/it/transaction_hash_fetching.rs
Comment (mattsse) [inline crates/net/network/tests/it/transaction_hash_fetching.rs:2]: this isn't needed because we're already in the tests scope

**Comment by Elvis339** [comment]
Just bumping this up for visibility.

**Comment by mattsse** [comment]
nope this is good, was just holding off in case we introduced some conflicts with some recent changes,

looks like we're good

**Review by mattsse — APPROVED** [review]

**Pull request merged**
